app/services/authentic_jobs_service.rb

```diff
@@ -0,0 +1,83 @@
+class AuthenticJobsService < JobFetcher
+  def self.scrape(term)
+    jobs = self.get_jobs(term)
+    self.format_and_create_entries(jobs, term)
+  end
+
+  def self.format_and_create_entries(entries, term)
+    entries.each do |entry|
+      formatted_entry = self.format_entry(entry, term)
+      self.create_records(formatted_entry) unless entry[:company].nil?
+    end
+  end
+
+  def self.format_entry(entry,term)
+    { job: {
+        title: self.full_title(entry),
+        url: entry[:url],
+        location: self.pull_location(entry),
+        raw_technologies: self.pull_technologies(entry[:description], term),
+        description: entry[:description],
+        remote: entry[:telecommuting] == 1 ? true : false,
+        posted_date: entry[:post_date]
+      },
+      company: {
+        name: self.pull_company_name(entry)
+      }
+    }
+  end
+
+  def self.full_title(entry)
+    if !entry[:company].nil?
+      "#{entry[:title]} at #{self.pull_company_name(entry)} (#{self.pull_location(entry)})"
+    else
+      entry[:title]
+    end
+  end
+
+  def self.pull_location(entry)
+    if !entry[:company].nil? && !entry[:company][:location].nil?
+      entry[:company][:location][:name]
+    elsif entry[:telecommuting] == 1
+      "Remote"
+    else
+      ""
+    end
+  end
+
+  def self.pull_company_name(entry)
+    entry[:company].nil? ? "" : entry[:company][:name]
+  end
+
+  def self.pull_technologies(description, term)
+    tech = technologies.select do |tech|
+      regex = /\b#{tech}\b/i
+      regex.match(description)
+    end
+    tech.empty? ? [term] : tech
+  end
+
+  def self.get_jobs(term)
```

Comment:
I wonder if you may want to split these three methods (get_jobs, connect_to_authentic_jobs, initial_connection) a bit differently. Seems there are three discrete tasks: "get_jobs", which is made up of "establish_connection" and "request_jobs_meeting_criteria". "get_jobs" would look like: 

```
connection = establish_connection
response = self.parse(request_jobs_meeting_criteria(connection, terms))
response[:listings].nil? ? [] : response[:listings][:listing]
```

and "request_jobs_meeting_criteria" would be: 
```
connection.params['api_key'] = ENV['AUTHENTIC_JOBS_KEY']
connection.params['format'] = 'json'
connection.params['method'] = 'aj.jobs.search'
connection.params['perpage'] = '100'
connection.params['keywords'] = term
connection
```
It's pretty readable already, but this may be a nice tweak.